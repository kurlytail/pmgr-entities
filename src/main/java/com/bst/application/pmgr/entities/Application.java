package com.bst.application.pmgr.entities;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.web.servlet.support.SpringBootServletInitializer;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.transaction.annotation.EnableTransactionManagement;

import com.bst.configuration.pmgr.entities.PmgrEntitiesConfiguration;

@SpringBootApplication
@ComponentScan(basePackageClasses = { PmgrEntitiesConfiguration.class, Application.class })
@EnableTransactionManagement
public class Application extends SpringBootServletInitializer {

	public static void main(final String[] args) {
		SpringApplication.run(Application.class, args);
	}
}