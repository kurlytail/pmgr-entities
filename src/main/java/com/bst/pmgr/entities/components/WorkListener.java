package com.bst.pmgr.entities.components;

import java.util.stream.Collectors;

import javax.persistence.PrePersist;

import org.kohsuke.randname.RandomNameGenerator;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import com.bst.pmgr.entities.Activity;
import com.bst.pmgr.entities.Document;
import com.bst.pmgr.entities.Process;
import com.bst.pmgr.entities.ProcessGroup;
import com.bst.pmgr.entities.Work;
import com.bst.pmgr.entities.schema.SchemaService;

@Component
public class WorkListener {

	private final RandomNameGenerator rnd = new RandomNameGenerator(0);

	@Autowired
	private SchemaService schemaService;

	@PrePersist
	public void prePersist(final Work work) throws Exception {
		if (work.getName() == null) {
			work.setName(this.rnd.next());
		}

		if (work.getActivities().size() == 0) {
			this.schemaService.getActivities().values().stream().map(metaActivity -> {
				final Activity activity = new Activity();
				activity.setMetaName(metaActivity.getName());
				activity.setName(this.rnd.next());
				work.addToActivities(activity);
				return activity;
			}).collect(Collectors.toList());
		}

		if (work.getDocuments().size() == 0) {
			this.schemaService.getDocuments().values().stream().map(metaDocument -> {
				final Document document = new Document();
				document.setMetaName(metaDocument.getName());
				document.setName(this.rnd.next());
				work.addToDocuments(document);
				return document;
			}).collect(Collectors.toList());
		}

		if (work.getProcesses().size() == 0) {
			this.schemaService.getProcesses().values().stream().map(metaProcess -> {
				final Process process = new Process();
				process.setMetaName(metaProcess.getName());
				process.setName(this.rnd.next());
				work.addToProcesses(process);
				return process;
			}).collect(Collectors.toList());
		}

		if (work.getProcessGroups().size() == 0) {
			this.schemaService.getProcessGroups().values().stream().map(metaProcessGroup -> {
				final ProcessGroup processGroup = new ProcessGroup();
				processGroup.setMetaName(metaProcessGroup.getName());
				processGroup.setName(this.rnd.next());
				work.addToProcessGroups(processGroup);
				return processGroup;
			}).collect(Collectors.toList());
		}
	}

}
