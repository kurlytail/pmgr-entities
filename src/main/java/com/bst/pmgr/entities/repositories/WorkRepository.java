package com.bst.pmgr.entities.repositories;

import org.springframework.data.repository.CrudRepository;
import org.springframework.data.rest.core.annotation.RepositoryRestResource;

import com.bst.pmgr.entities.Work;
import com.google.common.base.Optional;

@RepositoryRestResource
public interface WorkRepository extends CrudRepository<Work, Long>, WorkRepositoryStrategy {
	Optional<Work> findByName(final String name);
}
