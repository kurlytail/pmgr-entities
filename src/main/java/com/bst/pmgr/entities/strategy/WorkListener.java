package com.bst.pmgr.entities.strategy;

import java.util.stream.Collectors;

import javax.persistence.PrePersist;

import org.kohsuke.randname.RandomNameGenerator;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import com.bst.pmgr.entities.Activity;
import com.bst.pmgr.entities.Work;
import com.bst.pmgr.entities.schema.SchemaService;

@Component
public class WorkListener {

	private RandomNameGenerator rnd = new RandomNameGenerator(0);

	@Autowired
	private SchemaService schemaService;

	@PrePersist
	public void prePersist(final Work work) throws Exception {
		if (work.getName() == null) {
			work.setName(rnd.next());
		}

		if (work.getActivities().size() == 0) {
			work.getActivities().addAll(schemaService.getActivities().values().stream().map(metaActivity -> {
				Activity activity = new Activity();
				activity.setMetaName(metaActivity.getName());
				activity.setName(rnd.next());
				return activity;
			}).collect(Collectors.toList()));
		}
	}
}
