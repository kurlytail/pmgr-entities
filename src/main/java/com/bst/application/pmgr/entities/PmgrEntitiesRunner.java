package com.bst.application.pmgr.entities;

import java.util.Random;

import javax.persistence.EntityManager;
import javax.transaction.Transactional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Service;

import com.bst.pmgr.entities.Work;
import com.bst.pmgr.entities.repositories.WorkRepository;

@Service
@Transactional
public class PmgrEntitiesRunner implements CommandLineRunner {

	@Autowired
	private WorkRepository workRepository;

	@Autowired
	private EntityManager entityManager;

	@Override
	public void run(final String... args) throws Exception {

		final long countOfWork = Math.abs(new Random().nextLong()) % 10;
		long indexOfWork = 0;
		
		while (indexOfWork < countOfWork) {
			final Work work = new Work();
			this.workRepository.save(work);
			++indexOfWork;
		}

		this.entityManager.flush();
	}

}
