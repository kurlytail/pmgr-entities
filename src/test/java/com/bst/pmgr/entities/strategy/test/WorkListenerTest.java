package com.bst.pmgr.entities.strategy.test;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.autoconfigure.domain.EntityScan;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.EnableAspectJAutoProxy;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.TestExecutionListeners;
import org.springframework.test.context.TestExecutionListeners.MergeMode;
import org.springframework.test.context.junit.jupiter.SpringExtension;

import com.bst.pmgr.entities.Work;
import com.bst.pmgr.entities.components.WorkListener;
import com.bst.pmgr.entities.repositories.WorkRepository;
import com.bst.pmgr.entities.schema.SchemaService;
import com.bst.utility.components.RepositoryAspect;
import com.bst.utility.testlib.SnapshotListener;

@ExtendWith(SpringExtension.class)
@TestExecutionListeners(listeners = SnapshotListener.class, mergeMode = MergeMode.MERGE_WITH_DEFAULTS)
@DataJpaTest
@ContextConfiguration(classes = { TestConfiguration.class })
public class WorkListenerTest {

	@TestConfiguration
	@EntityScan("com.bst.pmgr.entities")
	@EnableJpaRepositories("com.bst.pmgr.entities.repositories")
	@EnableAspectJAutoProxy
	static class WorkListenerTestConfiguration {

		@Bean
		public WorkListener getWorkListener() {
			return new WorkListener();
		}
		
		@Bean
		public RepositoryAspect getRepositoryAspect() {
			return new RepositoryAspect();
		}

		@Bean
		public SchemaService getSchemaService() {
			return new SchemaService();
		}

	}

	@Autowired
	private WorkRepository workRepository;

	@SuppressWarnings("unused")
	@Autowired
	private WorkListener workListener;

	@Test
	public void testWorkListenerInitializeWork() throws Exception {
		Work work = new Work();
		work = this.workRepository.save(work);
		SnapshotListener.expect(work).toMatchSnapshot();
	}
}
