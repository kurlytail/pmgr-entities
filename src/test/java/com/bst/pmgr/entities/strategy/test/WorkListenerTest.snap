{
  "testWorkListenerInitializeWork" : [ {
    "id" : 1,
    "name" : "constructive_carrot",
    "tools" : [ ],
    "activities" : [ {
      "id" : 3,
      "name" : "flexible_designer",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "contractAdministration"
    }, {
      "id" : 4,
      "name" : "linear_fund",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "costEstimating"
    }, {
      "id" : 5,
      "name" : "popular_leaf",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "humanResourcePlanning"
    }, {
      "id" : 6,
      "name" : "steady_parent",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "qualitativeRiskAnalysis"
    }, {
      "id" : 7,
      "name" : "abstract_rest",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "planContracting"
    }, {
      "id" : 8,
      "name" : "controversial_supply",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "createWbs"
    }, {
      "id" : 9,
      "name" : "fragrant_absorption",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "directAndManageProjectExecution"
    }, {
      "id" : 10,
      "name" : "lively_cassette",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "monitorAndControlProjectWork"
    }, {
      "id" : 11,
      "name" : "powerful_destruction",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "performanceReporting"
    }, {
      "id" : 12,
      "name" : "straight_fuss",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "riskManagementPlanning"
    }, {
      "id" : 13,
      "name" : "accessible_lecture",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "costBudgeting"
    }, {
      "id" : 14,
      "name" : "cooperative_participant",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "quantitativeRiskAnalysis"
    }, {
      "id" : 15,
      "name" : "fresh_restriction",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "acquireProjectTeam"
    }, {
      "id" : 16,
      "name" : "loose_surgery",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "communicationsPlanning"
    }, {
      "id" : 17,
      "name" : "pregnant_acceptance",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "activitySequencing"
    }, {
      "id" : 18,
      "name" : "strong_cat",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "scheduleDevelopment"
    }, {
      "id" : 19,
      "name" : "addicted_deviation",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "contractClosure"
    }, {
      "id" : 20,
      "name" : "cruel_game",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "performQualityAssurance"
    }, {
      "id" : 21,
      "name" : "full-time_legend",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "requestSellerResponses"
    }, {
      "id" : 22,
      "name" : "loyal_party",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "developProjectCharter"
    }, {
      "id" : 23,
      "name" : "primary_retreat",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "manageStakeholders"
    }, {
      "id" : 24,
      "name" : "stupid_survivor",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "developPreliminaryProjectScopeStatement"
    }, {
      "id" : 25,
      "name" : "aggressive_accumulation",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "planPurchasesAndAcquisitions"
    }, {
      "id" : 26,
      "name" : "cute_cattle",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "riskMonitoringAndControl"
    }, {
      "id" : 27,
      "name" : "generous_dialogue",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "closeProject"
    }, {
      "id" : 28,
      "name" : "major_garden",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "scopeControl"
    }, {
      "id" : 29,
      "name" : "professional_lemon",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "integratedChangeControl"
    }, {
      "id" : 30,
      "name" : "sudden_passion",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "riskResponsePlanning"
    }, {
      "id" : 31,
      "name" : "alive_revival",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "activityResourceEstimating"
    }, {
      "id" : 32,
      "name" : "dead_sweet",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "activityDurationEstimating"
    }, {
      "id" : 33,
      "name" : "glad_acquisition",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "manageProjectTeam"
    }, {
      "id" : 34,
      "name" : "mathematical_celebration",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "scopeVerification"
    }, {
      "id" : 35,
      "name" : "proper_diet",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "costControl"
    }, {
      "id" : 36,
      "name" : "surprised_gear",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "scopePlanning"
    }, {
      "id" : 37,
      "name" : "ambitious_liability",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "selectSellers"
    }, {
      "id" : 38,
      "name" : "decorative_pasture",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "developProjectTeam"
    }, {
      "id" : 39,
      "name" : "grand_rhythm",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "scheduleControl"
    }, {
      "id" : 40,
      "name" : "mechanical_syllable",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "scopeDefinition"
    }, {
      "id" : 41,
      "name" : "public_addition",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "developProjectManagementPlan"
    }, {
      "id" : 42,
      "name" : "systematic_censorship",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "informationDistribution"
    }, {
      "id" : 43,
      "name" : "anonymous_dimension",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "activityDefinition"
    }, {
      "id" : 44,
      "name" : "democratic_gentleman",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "qualityPlanning"
    }, {
      "id" : 45,
      "name" : "green_lid",
      "tools" : [ ],
      "work" : 1,
      "metaName" : "riskIdentification"
    } ],
    "processes" : [ {
      "id" : 139,
      "name" : "false_pin",
      "work" : 1,
      "metaName" : "communicationsManagement"
    }, {
      "id" : 140,
      "name" : "joint_sand",
      "work" : 1,
      "metaName" : "timeManagement"
    }, {
      "id" : 141,
      "name" : "passionate_thesis",
      "work" : 1,
      "metaName" : "qualityManagement"
    }, {
      "id" : 142,
      "name" : "smart_ankle",
      "work" : 1,
      "metaName" : "scopeManagement"
    }, {
      "id" : 143,
      "name" : "warm_citizen",
      "work" : 1,
      "metaName" : "costManagement"
    }, {
      "id" : 144,
      "name" : "competent_drawing",
      "work" : 1,
      "metaName" : "integrationManagement"
    }, {
      "id" : 145,
      "name" : "fast_guest",
      "work" : 1,
      "metaName" : "procurementManagement"
    }, {
      "id" : 146,
      "name" : "kind_manner",
      "work" : 1,
      "metaName" : "humanResourceManagement"
    }, {
      "id" : 147,
      "name" : "peaceful_pitch",
      "work" : 1,
      "metaName" : "riskManagement"
    } ],
    "documents" : [ {
      "id" : 46,
      "name" : "middle-class_patience",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "staffingManagementPlan"
    }, {
      "id" : 47,
      "name" : "quiet_rider",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "enterpriseEnvironmentalFactors"
    }, {
      "id" : 48,
      "name" : "tasty_syndrome",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "selectedSellers"
    }, {
      "id" : 49,
      "name" : "archaeological_admission",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "validatedDefectRepair"
    }, {
      "id" : 50,
      "name" : "difficult_ceremony",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "resolvedIssues"
    }, {
      "id" : 51,
      "name" : "hard_director",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "wbs"
    }, {
      "id" : 52,
      "name" : "minor_giant",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "forecastedCompletion"
    }, {
      "id" : 53,
      "name" : "rational_lift",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "projectCalendar"
    }, {
      "id" : 54,
      "name" : "tense_pause",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "administrativeClosureProcedure"
    }, {
      "id" : 55,
      "name" : "ashamed_ring",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "procurementManagementPlan"
    }, {
      "id" : 56,
      "name" : "dirty_tail",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "qualityManagementPlan"
    }, {
      "id" : 57,
      "name" : "heavy_advantage",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "deliverables"
    }, {
      "id" : 58,
      "name" : "modest_chalk",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "riskManagementPlan"
    }, {
      "id" : 59,
      "name" : "realistic_disagreement",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "qualityMetrics"
    }, {
      "id" : 60,
      "name" : "thirsty_glance",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "performanceMeasurementBaseline"
    }, {
      "id" : 61,
      "name" : "attractive_limitation",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "implementedPreventiveActions"
    }, {
      "id" : 62,
      "name" : "domestic_peanut",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "costBaseline"
    }, {
      "id" : 63,
      "name" : "hilarious_ritual",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "scopeBaseline"
    }, {
      "id" : 64,
      "name" : "moral_tape",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "projectScopeStatement"
    }, {
      "id" : 65,
      "name" : "regular_adviser",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "approvedDefectRepair"
    }, {
      "id" : 66,
      "name" : "tired_chance",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "resourceAvailability"
    }, {
      "id" : 67,
      "name" : "aware_discount",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "wbsDictionary"
    }, {
      "id" : 68,
      "name" : "due_glory",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "qualityChecklists"
    }, {
      "id" : 69,
      "name" : "honest_lion",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "workPerformanceInformation"
    }, {
      "id" : 70,
      "name" : "musical_penalty",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "riskRegister"
    }, {
      "id" : 71,
      "name" : "reliable_robot",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "rejectedChangeRequests"
    }, {
      "id" : 72,
      "name" : "traditional_taxi",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "contract"
    }, {
      "id" : 73,
      "name" : "balanced_age",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "scheduleBaseline"
    }, {
      "id" : 74,
      "name" : "early_chapter",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "acceptedDeliverables"
    }, {
      "id" : 75,
      "name" : "hot_disease",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "implementedCorrectiveActions"
    }, {
      "id" : 76,
      "name" : "native_goal",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "milestoneList"
    }, {
      "id" : 77,
      "name" : "residential_literacy",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "resourceRequirements"
    }, {
      "id" : 78,
      "name" : "true_pensioner",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "projectScopeManagementPlan"
    }, {
      "id" : 79,
      "name" : "beautiful_roof",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "performanceReports"
    }, {
      "id" : 80,
      "name" : "effective_tear",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "preliminaryProjectScopeStatement"
    }, {
      "id" : 81,
      "name" : "ignorant_agony",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "approvedPreventiveActions"
    }, {
      "id" : 82,
      "name" : "neutral_charity",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "outsourceContract"
    }, {
      "id" : 83,
      "name" : "restricted_disorder",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "procurementDocumentPackage"
    }, {
      "id" : 84,
      "name" : "unanimous_gold",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "projectManagementProcesses"
    }, {
      "id" : 85,
      "name" : "blank_loan",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "proposals"
    }, {
      "id" : 86,
      "name" : "eligible_performance",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "activityCostEstimates"
    }, {
      "id" : 87,
      "name" : "implicit_rose",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "recommendedCorrectiveActions"
    }, {
      "id" : 88,
      "name" : "noisy_telephone",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "forecasts"
    }, {
      "id" : 89,
      "name" : "right_aids",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "resourceCalendars"
    }, {
      "id" : 90,
      "name" : "unemployed_chauvinist",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "qualityControlMeasurements"
    }, {
      "id" : 91,
      "name" : "bold_distributor",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "scheduleManagementPlan"
    }, {
      "id" : 92,
      "name" : "empty_gown",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "projectOrganizationalCharts"
    }, {
      "id" : 93,
      "name" : "inadequate_log",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "closedContracts"
    }, {
      "id" : 94,
      "name" : "obese_person",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "processImprovementPlan"
    }, {
      "id" : 95,
      "name" : "round_routine",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "evaluationCriteria"
    }, {
      "id" : 96,
      "name" : "uniform_temptation",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "projectScheduleNetworkDiagrams"
    }, {
      "id" : 97,
      "name" : "bureaucratic_aisle",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "finalProductServiceOrResult"
    }, {
      "id" : 98,
      "name" : "environmental_chemistry",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "activityDurationEstimates"
    }, {
      "id" : 99,
      "name" : "incredible_division",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "approvedChangeRequests"
    }, {
      "id" : 100,
      "name" : "occupational_grain",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "projectFundingRequirements"
    }, {
      "id" : 101,
      "name" : "safe_loss",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "makeBuyDecisions"
    }, {
      "id" : 102,
      "name" : "unlike_phenomenon",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "requestedChanges"
    }, {
      "id" : 103,
      "name" : "cautious_ruin",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "projectSchedule"
    }, {
      "id" : 104,
      "name" : "eternal_tent",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "activityAttributes"
    }, {
      "id" : 105,
      "name" : "indoor_allocation",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "organizationalProcessAssets"
    }, {
      "id" : 106,
      "name" : "old_chicken",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "rolesAndResponsibilities"
    }, {
      "id" : 107,
      "name" : "seasonal_dog",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "activityList"
    }, {
      "id" : 108,
      "name" : "useful_grant",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "scheduleModelData"
    }, {
      "id" : 109,
      "name" : "charismatic_loyalty",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "contractClosureProcedure"
    }, {
      "id" : 110,
      "name" : "exact_photography",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "qualityBaseline"
    }, {
      "id" : 111,
      "name" : "influential_rush",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "projectStatementOfWork"
    }, {
      "id" : 112,
      "name" : "optimistic_terrorist",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "costManagementPlan"
    }, {
      "id" : 113,
      "name" : "senior_aluminium",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "implementedDefectRepair"
    }, {
      "id" : 114,
      "name" : "vague_chimney",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "projectStaffAssignments"
    }, {
      "id" : 115,
      "name" : "chronic_domination",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "teamPerformanceAssessment"
    }, {
      "id" : 116,
      "name" : "exotic_gravel",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "resourcesBreakdownStructure"
    }, {
      "id" : 117,
      "name" : "insufficient_machine",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "recommendedDefectRepair"
    }, {
      "id" : 118,
      "name" : "organic_pie",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "projectCharter"
    }, {
      "id" : 119,
      "name" : "sexual_sailor",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "contractDocumentation"
    }, {
      "id" : 120,
      "name" : "varied_thanks",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "projectManagementPlan"
    }, {
      "id" : 121,
      "name" : "clean_amendment",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "approvedCorrectiveActions"
    }, {
      "id" : 122,
      "name" : "experimental_chord",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "costEstimate"
    }, {
      "id" : 123,
      "name" : "interactive_double",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "activityResourceRequirements"
    }, {
      "id" : 124,
      "name" : "outer_grief",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "procurementDocuments"
    }, {
      "id" : 125,
      "name" : "shy_maid",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "recommendedPreventiveActions"
    }, {
      "id" : 126,
      "name" : "vicious_pigeon",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "contractManagementPlan"
    }, {
      "id" : 127,
      "name" : "cold_salt",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "performanceMeasurements"
    }, {
      "id" : 128,
      "name" : "extinct_theorist",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "riskRelatedContractualAgreements"
    }, {
      "id" : 129,
      "name" : "international_angel",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "qualifiedSellerList"
    }, {
      "id" : 130,
      "name" : "paralyzed_cinema",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "activityCostEstimateSupportingDetail"
    }, {
      "id" : 131,
      "name" : "skilled_draft",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "communicationsManagementPlan"
    }, {
      "id" : 132,
      "name" : "visual_grounds",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "implementedChangeRequests"
    }, {
      "id" : 133,
      "name" : "commercial_makeup",
      "work" : 1,
      "sections" : [ ],
      "topLevelSections" : [ ],
      "metaName" : "contractStatementOfWork"
    } ],
    "processGroups" : [ {
      "id" : 134,
      "name" : "soft_sauce",
      "work" : 1,
      "metaName" : "monitoringAndControlling"
    }, {
      "id" : 135,
      "name" : "well_thread",
      "work" : 1,
      "metaName" : "planning"
    }, {
      "id" : 136,
      "name" : "comprehensive_anticipation",
      "work" : 1,
      "metaName" : "closing"
    }, {
      "id" : 137,
      "name" : "federal_clash",
      "work" : 1,
      "metaName" : "initiating"
    }, {
      "id" : 138,
      "name" : "latest_drill",
      "work" : 1,
      "metaName" : "executing"
    } ],
    "person" : {
      "id" : 2,
      "name" : "test person",
      "person" : { },
      "works" : [ ]
    }
  } ]
}